controllers.py

```diff
@@ -70,41 +76,47 @@ def alterar_genero(id):
     genero = genero_from_web(**request.json)
     if valida_id(id):
         update_genero(id, **genero)
-        return jsonify(genero_from_db(genero))
+        genero_alterado = get_genero(genero["nome"])
+        return jsonify(generos_from_db(genero_alterado))
     else:
         return jsonify({"erro": "Genêro Inválido."})
 
+
 @app.route("/filmes/<int:id>", methods=["PUT"])
 def alterar_filme(id):
     filme = filme_from_web(**request.json)
     if valida_id(id):
         update_filme(id, **filme)
-        return jsonify(filme_from_db(filme))
+        filme_alterado = get_filme("titulo", "ano", "classificacao", "preco", "diretores_id", "generos_id")
+        return jsonify(filmes_from_db(filme_alterado))
     else:
         return jsonify({"erro": "Filme Inválido."})
 
+
 @app.route("/usuarios/<int:id>", methods=["PUT"])
 def alterar_usuario(id):
     usuario = usuario_from_web(**request.json)
     if valida_id(id):
         update_usuario(id, **usuario)
-        return jsonify(usuario_from_db(usuario))
+        return jsonify(usuarios_from_db(usuario))
     else:
         return jsonify({"erro": "Usuário Inválido."})
 
 
 # Apagar => Diretor, Gênero, Filme, Usuário
 # Apagar => requisitar delete na url que tem o id do item, se tiver chave estrangeira,
 # exibe uma mensagem de erro tratado
+
 @app.route("/diretores/<int:id>", methods=["DELETE"])
 def apagar_diretor(id):
     diretor_id = delete_id_from_web(**request.json)
     try:
         if valida_id(id):
             delete_diretor(**diretor_id)
-            return jsonify(delete_id_from_db(diretor_id))
+            return jsonify(delete_id_from_db("", 204))
```

Comment:
acho que o aqui era só
```suggestion
            return None, 204
```